controller/model/posture_check_model_os_test.go

```diff
@@ -0,0 +1,304 @@
+		osCheck.OperatingSystems[0].OsVersions[0] = ">=10.5.19042" //default data has 10.5.19041
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns true for valid os type exactly matching version or higher", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		osCheck.OperatingSystems[0].OsVersions[0] = ">=10.5.19041" //default data has 10.5.19041
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns false for invalid os but exactly matching version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		osCheck.OperatingSystems[0].OsType = os2Type
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for valid os and no matching version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = ""
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns true for valid os and no required version and no submitted version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os3Type
+		postureData.Os.Version = ""
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for valid os and no required version and submitted version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os3Type
+		postureData.Os.Version = "1.2.3"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for lower exact version match and later and higher match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = os2Version1
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for higher exact version match and an earlier match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = "7.8.9"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	//poorly formatted posture data
+	t.Run("returns false for posture data with valid os and partial os major version match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for posture data with valid os and partial os major and minor version match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10.5"
```

Comment:
mac os might be 14.1 and 14.2 ? is the minor really necessary? ==> 14.1.0 ,14.2.0 ?

Reply:
I looked up their versions and they have a patch number.

Reply:
https://stackoverflow.com/questions/157759/how-can-i-determine-the-running-mac-os-x-version-programmatically

Reply:
yes - mac has a patch number.  I'm not currently sending it up since MOP isn't configured for it, and this (version x.y.z of higher") wasn't yet supported.

iOS doesn't supply a patch number in the version.

Reply:
We should send in 0 where it doesn't exist. Otherwise, I have to introduce special cases.

Reply:
okay, no problem

Reply:
😎